autoload/ale/lsp_linter.vim

```diff
@@ -271,6 +271,20 @@ function! ale#lsp_linter#OnInit(linter, details, Callback) abort
         call ale#lsp#NotifyForChanges(l:conn_id, l:buffer)
     endif
 
+    " Tell the relevant buffer that the LSP has started.
+    if l:buffer > 0 && l:buffer == bufnr('')
```

Comment:
This `l:buffer > 0` check might not be relevant / useful here.

Reply:
Probably neither branch should run if the buffer number is invalid.